### DISCUSSION TITLE
关于cache-persist的疑问

### DISCUSSION THREAD

**Discussion by yyysuo** [pymumu/smartdns#1097] [General]
1：cache-persist默认的路径是tmp，重启就没有了persist指的是什么呢
2：把cache-persist的路径设置到/root，重启后cache-persist文件还是重新生成了，所以重启之前的缓存还是会被丢弃么。

希望：
1：cache-persist在路由重启时也可以自动保存，永久有效。
2：不抛弃任何冷数据，建立所有访问过域名的缓存，并可以按天定时更新。

**Comment by PikuZheng** [comment]
持久缓存文件在smartdns正常退出时写入文件。查看smartdns启动和关闭时的日志可以看到缓存有多少条。

根据RFC规范，DNS数据由起始授权机构提供有效期（一般是300秒到3600秒），在有效期过后有必要重新递归查询直到获得最新数据。将DNS数据一直缓存不变不符合规范（DDNS有话要说）。

**Reply by yyysuo** [reply]
后台不是会自动按每分钟xx条的速度进行更新么？

**Reply by PikuZheng** [reply]
没有这种事

**Reply by PikuZheng** [reply]
虽然缓存可能在不断更新，但可能具体行为和你的理解有相当大的差异。建议阅读配置说明 prefetch 相关
